lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,60 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    collection.group_by(&:class)[Fixnum].count + collection.group_by(&:class)[Float].count
   end
 
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
   def all_words_longer_than_length?(min_length)
+    collection.all? { |x| x.length > min_length }
```

Comment:
Best practice is to try to use descriptive names rather than `|x|`. In this case that might be `|element|` or `|item|`.